src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy

```diff
@@ -3716,14 +3716,24 @@ public enum DarknessAblaze implements LogicCardInfo {
       return itemCard (this) {
         text "Flip 2 coins. If both are heads, choose 1 card from your discard pile, reveal it, and put it into your hand."
         onPlay {
+          def heads = 2
+          flip 2, {}, {heads -= 1}
+
+          if (heads == 2) {
```

Comment:
```suggestion
          flip 2,{},{},[2:{my.discard.select("Which card will you put back into your hand?").move(my.hand)}]
``` is a cleaner way to do this

Reply:
Personally find the if cleaner to read, without added complexity as far as I can see.

Reply:
I could see this method being cleaner if you defined the closure outside of the map, and just provided it to the map.
```Groovy
def fromDiscard = {my.discard.select("Which card will you put back into your hand?").move(my.hand)}
flip 2, {}, {}, [2:fromDiscard]
```